rules-java/api/src/main/resources/reports/templates/report_index.ftl

```diff
@@ -209,16 +209,18 @@
     <script type="text/javascript">
         var tbodyElement = $("#javaIncidentsByPackageTBody");
 
-        var rows = "";
-        for (var i = 0; i < WINDUP_PACKAGE_PIE_DATA['application_pie'].length; i++) {
-            var row = "";
-            row += "<tr>";
-            row += "<td>" + WINDUP_PACKAGE_PIE_DATA['application_pie'][i].label + "</td>";
-            row += "<td class='numeric-column'>" + WINDUP_PACKAGE_PIE_DATA['application_pie'][i].data + "</td>";
-            row += "</tr>";
-            rows += row;
+        if (typeof(WINDUP_PACKAGE_PIE_DATA) !== 'undefined') {
+            var rows = "";
+            for (var i = 0; i < WINDUP_PACKAGE_PIE_DATA['application_pie'].length; i++) {
+                var row = "";
+                row += "<tr>";
+                row += "<td>" + WINDUP_PACKAGE_PIE_DATA['application_pie'][i].label + "</td>";
+                row += "<td class='numeric-column'>" + WINDUP_PACKAGE_PIE_DATA['application_pie'][i].data + "</td>";
+                row += "</tr>";
+                rows += row;
+            }
+            tbodyElement.prepend(rows);
         }
```

Comment:
Yes. This keeps it from rendering a broken section (along with a JavaScript error) in the case that there is no data for this section..